dist/tools/ethos/ethos.c

```diff
@@ -49,8 +52,18 @@ static void usage(void)
 
 static void checked_write(int handle, void *buffer, int nbyte)
 {
-    if (write(handle, buffer, nbyte) != nbyte) {
-        fprintf(stderr, "write to fd %i failed: %s\n", handle, strerror(errno));
+    ssize_t res;
+    while (nbyte > 0) {
+        res = write(handle, buffer, nbyte);
+        if (res < 0) {
```

Comment:
why two seperate if-cases? looks to me like one error message should be enough, with "res=%i".